lib/msal-core/test/UserAgentApplication.spec.ts

```diff
@@ -580,17 +610,21 @@ describe("UserAgentApplication.ts Class", function () {
         it("exits login function with error if interaction is true", function (done) {
             cacheStorage.setItem(TemporaryCacheKeys.INTERACTION_STATUS, Constants.inProgress);
             window.location = oldWindowLocation;
+            let testsExecuted = false;
             const checkErrorFromLibrary = function (authErr: AuthError) {
                 expect(authErr instanceof ClientAuthError).to.be.true;
                 expect(authErr.errorCode).to.equal(ClientAuthErrorMessage.loginProgressError.code);
                 expect(authErr.errorMessage).to.equal(ClientAuthErrorMessage.loginProgressError.desc);
                 expect(authErr.message).to.equal(ClientAuthErrorMessage.loginProgressError.desc);
                 expect(authErr.name).to.equal("ClientAuthError");
                 expect(authErr.stack).to.include("UserAgentApplication.spec.ts");
+                testsExecuted = true;
```

Comment:
The `done` function should be used for this, why is a boolean needed?

Reply:
This is to ensure the callback is actually hit and the tests are run. I was running into an issue where all the tests were passing when they should have been failing. Figured out the callback was never being run and thus no tests to pass or fail.

Reply:
Interesting, I thought if you passed `done` the test would timeout if `done` wasn't called...

Reply:
Oh you know what, you're totally right. This was happening on tests that were not marked as async and I just applied this to everything calling ```handleRedirectCallback``` without realizing the difference. So I guess the solution here maybe is to make those other tests async instead of doing this

Reply:
@tnorling Ah, yeah using `done` for those tests should fix.